x/upgrade/keeper/keeper.go

```diff
@@ -161,6 +163,35 @@ func (k Keeper) GetUpgradedClient(ctx sdk.Context) (ibcexported.ClientState, int
 	return clientState, int64(height), nil
 }
 
+// SetUpgradedConsensusState set the expected upgraded consensus state for the next version of this chain
+// using the last height committed on this chain.
+func (k Keeper) SetUpgradedConsensusState(ctx sdk.Context, lastHeight int64, cs ibcexported.ConsensusState) error {
+	store := ctx.KVStore(k.storeKey)
+	bz, err := clienttypes.MarshalConsensusState(k.cdc, cs)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "could not marshal consensus state: %v", err)
+	}
+
+	store.Set(types.UpgradedConsStateKey(lastHeight), bz)
+	return nil
+}
+
+// GetUpgradedConsensusState set the expected upgraded consensus state for the next version of this chain
+// NOTE: Used for testing purposes only
+func (k Keeper) GetUpgradedConsensusState(ctx sdk.Context, lastHeight int64) (ibcexported.ConsensusState, error) {
```

Comment:
if this is used for testing only we should move it to ibctesting

Reply:
It's used in upgrade module testing. Since it's a getter for the setter in this file, I think it's fine to leave here for now.